tests/config_test.go

```diff
@@ -574,6 +578,242 @@ var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:comp
+					"option1": "value1",
+					"option2": "value2",
+					"option3": "value3",
+				}
+				tests.CreateConfigMap(configMapName, testsuite.NamespacePrivileged, data)
+			})
+
+			AfterEach(func() {
+				tests.DeleteConfigMap(configMapName, testsuite.NamespacePrivileged)
+			})
+
+			It("Should be the mounted virtiofs layout the same for a pod and vmi", func() {
+				expectedOutput := "value1value2value3"
+
+				By("Running VMI")
+				vmi := libvmi.NewFedora(
+					withConfigMapFs(configMapName, configMapName),
+					libvmi.WithNamespace(testsuite.NamespacePrivileged),
+				)
+				vmi = tests.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 300)
+				Eventually(matcher.ThisVMI(vmi), 12*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+
+				By("Logging into the VMI")
+				Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+				By("Checking if ConfigMap has been attached to the pod")
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				podOutput, err := exec.ExecuteCommandOnPod(
+					virtClient,
+					vmiPod,
+					fmt.Sprintf("virtiofs-%s", configMapName),
+					[]string{"cat",
+						configMapPath + "/option1",
+						configMapPath + "/option2",
+						configMapPath + "/option3",
+					},
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podOutput).To(Equal(expectedOutput))
+
+				By("Checking mounted ConfigMap")
+				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: fmt.Sprintf("mount -t virtiofs %s /mnt \n", configMapName)},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: "echo $?\n"},
+					&expect.BExp{R: console.RetValue("0")},
+					&expect.BSnd{S: "cat /mnt/option1 /mnt/option2 /mnt/option3\n"},
+					&expect.BExp{R: expectedOutput},
+				}, 200)).To(Succeed())
+			})
+		})
+
+		Context("With a single Secret volume", func() {
+			var (
+				secretName string
+				secretPath string
+			)
+
+			BeforeEach(func() {
+				// We use the Secret name as mount `tag` for qemu, but the `tag` property must be 36 bytes or less
+				secretName = "secret-" + uuid.NewRandom().String()[:6]
+				secretPath = config.GetSecretSourcePath(secretName)
+
+				data := map[string]string{
+					"user":     "admin",
+					"password": "redhat",
+				}
+				tests.CreateSecret(secretName, testsuite.NamespacePrivileged, data)
+			})
+
+			AfterEach(func() {
+				tests.DeleteSecret(secretName, testsuite.NamespacePrivileged)
+			})
+
+			It("Should be the mounted virtiofs layout the same for a pod and vmi", func() {
+				expectedOutput := "adminredhat"
+
+				By("Running VMI")
+				vmi := libvmi.NewFedora(
+					withSecretFs(secretName, secretName),
+					libvmi.WithNamespace(testsuite.NamespacePrivileged),
+				)
+				vmi = tests.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 300)
+				Eventually(matcher.ThisVMI(vmi), 12*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
```

Comment:
Why is the agent necessary?

Reply:
You are right, we don't need it. We were using it just to be sure the VM is up and running, but it's redundant since `Expect(console.LoginToFedora(vmi)).To(Succeed())` will do the same but with a smaller timeout. I'll remove it